test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -307,6 +307,66 @@ TEST_F(HttpFilterTest, ErrorFailClose) {
   EXPECT_EQ(1U, config_->stats().error_.value());
 }
 
+// Test when failure_mode_allow is NOT set and the response from the authorization service is Error, but the runtime guard set to false
+// that the request is not allowed to continue and should forward error status code and body.
+TEST_F(HttpFilterTest, ErrorFailCloseWithRuntimeGuardFalse) {
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues(
+      {{"envoy.reloadable_features.ext_authz_http_service_5xx_is_denied_instead_of_error",
+        "false"}});
+
+  InSequence s;
+
+  initialize(R"EOF(
+  transport_api_version: V3
+  grpc_service:
+    envoy_grpc:
+      cluster_name: "ext_authz_server"
+  failure_mode_allow: false
+  )EOF");
+
+  Filters::Common::ExtAuthz::Response response{};
+  response.status = Filters::Common::ExtAuthz::CheckStatus::Error;
+  response.status_code = Http::Code::InternalServerError;
+  response.body = std::string{"foo"};
+  
+  Filters::Common::ExtAuthz::ResponsePtr response_ptr =
+      std::make_unique<Filters::Common::ExtAuthz::Response>(response);  
+
+  prepareCheck();
+
+  EXPECT_CALL(*client_, check(_, _, _, _))
+      .WillOnce(
+          Invoke([&](Filters::Common::ExtAuthz::RequestCallbacks& callbacks,
+                     const envoy::service::auth::v3::CheckRequest&, Tracing::Span&,
+                     const StreamInfo::StreamInfo&) -> void {
+        request_callbacks_ = &callbacks;         
+      }));
+
+  EXPECT_EQ(Http::FilterHeadersStatus::StopAllIterationAndWatermark,
+            filter_->decodeHeaders(request_headers_, false));
+  EXPECT_CALL(decoder_filter_callbacks_, continueDecoding()).Times(0);
+  EXPECT_CALL(decoder_filter_callbacks_, encodeHeaders_(_, _))
```

Comment:
We need to "copy" the test in "common":

Here we need to expect:

```cpp
EXPECT_CALL(decoder_filter_callbacks_,
            encodeHeaders_(HeaderMapEqualRef(&response_headers), false))
```

Where `response_headers` is something like:

```cpp
Http::TestResponseHeaderMapImpl response_headers{
    {":status", "500"},
    // The "content-length" and "content-type" headers are here in the response headers since
    // authz service sends a non-empty body response.
    {"content-length", "4"},
    {"content-type", "text/plain"},
    {"foo", "bar"},
    {"x-foobar", "bar"}};
```